public/pages/AnomalyCharts/containers/AnomalyDetailsChart.tsx

```diff
@@ -0,0 +1,388 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import {
+  AnnotationDomainTypes,
+  Axis,
+  Chart,
+  LineAnnotation,
+  LineSeries,
+  niceTimeFormatter,
+  Position,
+  RectAnnotation,
+  ScaleType,
+  Settings,
+  XYBrushArea,
+} from '@elastic/charts';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiLoadingChart,
+  EuiStat,
+} from '@elastic/eui';
+import { get } from 'lodash';
+import moment from 'moment';
+import React, { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { useDelayedLoader } from '../../../hooks/useDelayedLoader';
+import {
+  AnomalySummary,
+  DateRange,
+  Detector,
+  Monitor,
+  MonitorAlert,
+} from '../../../models/interfaces';
+import { searchAlerts } from '../../../redux/reducers/alerting';
+import { darkModeEnabled } from '../../../utils/kibanaUtils';
+import {
+  filterWithDateRange,
+  prepareDataForChart,
+} from '../../utils/anomalyResultUtils';
+import { AlertsFlyout } from '../components/AlertsFlyout/AlertsFlyout';
+import {
+  AlertsStat,
+  AnomalyStatWithTooltip,
+} from '../components/AnomaliesStat/AnomalyStat';
+import {
+  convertAlerts,
+  disabledHistoryAnnotations,
+  generateAlertAnnotations,
+  getAnomalySummary,
+} from '../utils/anomalyChartUtils';
+import {
+  ANOMALY_CHART_THEME,
+  CHART_FIELDS,
+  INITIAL_ANOMALY_SUMMARY,
+} from '../utils/constants';
+import { HeatmapCell } from './AnomalyHeatmapChart';
+
+interface AnomalyDetailsChartProps {
+  onDateRangeChange(
+    startDate: number,
+    endDate: number,
+    dateRangeOption?: string
+  ): void;
+  onZoomRangeChange(startDate: number, endDate: number): void;
+  anomalies: any[];
+  bucketizedAnomalies: boolean;
+  anomalySummary: any;
+  dateRange: DateRange;
+  isLoading: boolean;
+  showAlerts?: boolean;
+  anomalyGradeSeriesName: string;
+  confidenceSeriesName: string;
```

Comment:
Minor: maybe combine this into a single `isSample` param or something, where the anomaly grade and anomaly confidence title names can be declared and maintained within the AnomalyDetailsChart itself, rather than having to pass in different hardcoded constants from other components

Reply:
In fact, looks like `showAlerts` prop already does this - could either reuse this prop to determine the title names and/or rename `showAlerts` to something like `isSample`

Reply:
sure. good idea. I agree for now we can just reuse `showAlerts` to do that.

Reply:
refactored this.